experimental/packages/opentelemetry-api-metrics/test/types/Metric.test.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Counter, UpDownCounter, Histogram } from "../../src";
+
+describe("Metric", () => {
+    describe("Counter", () =>{
+        it("enable not to define any type", () => {
+            const counter: Counter = {
+                add(_value: number, _attribute: unknown) {}
+            } as Counter
+            counter.add(1, { "some-attribute": "value" })
+        })
+
+        it("enable to use with type", () => {
+            type Attributes = {
+                "some-attribute": string
+            }
+            const counter: Counter<Attributes> = {
+                add(_value: number, _attribute: Attributes) {}
+            } as Counter<Attributes>
+            counter.add(1, { "some-attribute": "value" })
+        })
+
+        it("disable wrong attributes by typing", () => {
+            type Attributes = {
+                "some-attribute": string
+            }
+            const counter: Counter<Attributes> = {
+                add(_value: number, _attribute: Attributes) {}
+            } as Counter<Attributes>
```

Comment:
Nit: you don't need to cast the type of the literal objects if you have already declared the type of the variables.

```suggestion
            }
```